compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -975,12 +977,37 @@ impl<'interner> Monomorphizer<'interner> {
             }
 
             HirType::Struct(def, args) => {
+                // Even though we later call `convert_type` on `fields`, it might be the types
+                // in `args` end up not being part of fields. For example:
+                //
+                //     struct Foo<let N: u32> {}
+                //
+                //     fn main() {
+                //         let _ = Foo {};
+                //     }
+                //
+                // In the above case args is `[N]` but fields is `[]`, so T will never be checked.
+                // However, we want the above program to not compile.
+                for arg in args {
+                    if let Some(error) = Self::check_type(arg, location) {
+                        return Err(error);
+                    }
```

Comment:
I'd rather `check_type` returned a `Result` to begin with so that this could be:
```suggestion
                    Self::convert_type(arg, location)?;
```
(using convert_type here because of the next suggestion)

Reply:
Cool! I had it like `Result<(), ...>` in the beginning but I wasn't sure if it was okay to do it. But now that I remember, that's also how `fmt` works.